fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationsDialogBinding.kt

```diff
@@ -77,29 +77,35 @@ class TranslationsDialogBinding(
 
                 // Session Translations State Behavior (Tab)
                 val sessionTranslationsState = state.sessionState.translationsState
+
                 val fromSelected =
                     sessionTranslationsState.translationEngineState?.initialFromLanguage(
                         translateFromLanguages,
                     )
-
-                fromSelected?.let {
-                    translationsDialogStore.dispatch(
-                        TranslationsDialogAction.UpdateFromSelectedLanguage(
-                            fromSelected,
-                        ),
-                    )
+                // Dispatch initialFrom Language only the first time when it is null.
+                if (translationsDialogStore.state.initialFrom == null) {
+                    fromSelected?.let {
```

Comment:
```suggestion
                if (fromSelected != null && translationsDialogStore.state.initialFrom == null) {
```
Maybe we should skip a subsequent check here if we just did it all in the if statement conditional

Reply:
DOne